modules/ROOT/pages/access-source-message-attributes.adoc

```diff
@@ -0,0 +1,250 @@
+
+----
+POPurpose: 
+substringBefore(vars.b2bMessageContext.sourceEndpointAttributes.fileName, "_") default "PO-CREATE",
+----
+
+This example shows how to perform a `substringBefore` operation on the source filename to get the value before `_` and use it as the partner reference identifier:
+
+----
+%dw 2.0
+output application/json
+import * from dw::core::Strings
+ns ns0 http://xmlns.mulesoft.com/enterpriseobjects/b2b/
+---
+{
+	partnerReferenceId: substringBefore(vars.b2bMessageContext.sourceEndpointAttributes.fileName, "_") default "UNKNOWN" ,
+	hostReferenceId: payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#Seller,
+	businessDocumentKey: payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#PurchaseOrderNumber,
+	customAttributes: [
+		{
+			alias: "purchaseOrderNumber",
+			values: [payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#PurchaseOrderNumber]
+		},
+		{
+			alias: "buyerEmail",
+			values: [payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#BuyerEmailID]
+		}	]
+}
+----
+
+This example shows how to:
+
+* Use the partner identifier associated with an FTP or SFTP Receive from Partners endpoint configuration as the partner reference identifier for routing the message.
+* Map the source filename as a custom message attribute.
+
+----
+%dw 2.0
+output application/json
+ns ns0 http://xmlns.mulesoft.com/enterpriseobjects/b2b/
+---
+{
+	partnerReferenceId: vars.b2bMessageContext.sourceEndpointAttributes.partnerReferenceId,
+	hostReferenceId: payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#Seller,
+	businessDocumentKey: payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#PurchaseOrderNumber,
+	customAttributes: [
+		{
+			alias: "purchaseOrderNumber",
+			values: [payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#PurchaseOrderNumber]
+		},
+		{
+			alias: "buyerEmail",
+			values: [payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#BuyerEmailID]
+		},
+		{
+			alias: "sourceFileName",
+			values: [vars.b2bMessageContext.sourceEndpointAttributes.fileName default “UNKNOWN]
+		}
+	]
+}
+----
+
+This example shows how to map a custom header on the received HTTP message as a custom message attribute:
+
+----
+%dw 2.0
+output application/json
+ns ns0 http://xmlns.mulesoft.com/enterpriseobjects/b2b/
+---
+{
+	partnerReferenceId: payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#Seller,
+	hostReferenceId: payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#Buyer,
+	businessDocumentKey: payload.ns0#B2BMessage.ns0#Data.ns0#PurchaseOrder.ns0#PurchaseOrderNumber,
+	customAttributes: [
+		{
+			alias: "purchaseOrderNumber", 
+			values: payload.ns0#B2BMessage.ns0#Data.*ns0#PurchaseOrder map ( purchaseOrder , indexOfPurchaseOrder ) 
+						-> purchaseOrder.ns0#PurchaseOrderNumber default "UNKNOWN"
+		},
+		{
+			alias: "supplierName", 
+			values: [vars.b2bMessageContext.customProtocolHeaders.'x-apm-supplierName' default "UNKNOWN"]
+		}
+	]
+}
+----
+
+== See Also
+
+* xref:partner-manager-maps.adoc[Translation Maps]
```

Comment:
If you leave out the title from the [ ], it gets picked up automatically and you don't have to maintain the xrefs if the titles change.